planex/spec.py

```diff
@@ -253,14 +274,33 @@ def extract_source(self, name, destdir):
             target_path = os.path.normpath(os.path.join(self.prefix, name))
             tarball.extract(target_path, destdir)
 
+    def extract_sources(self, names, destdir):
+        """
+        Extract all the source [names] to [destdir].   Raises [KeyError]
+        for the first requested source that cannot be found.
+        """
+        with Tarball(self.path) as tarball:
+            for name in names:
+                target_path = os.path.normpath(os.path.join(self.prefix, name))
+                tarball.extract(target_path, destdir)
```

Comment:
https://docs.python.org/2/library/tarfile.html#tarfile.TarFile.extractall might be worth a look here. You would call getmembers(), take the intersection of that and the list you are looking for, then pass that intersection to extractall().

Reply:
When I looked at `extractall` I was worried by

> Warning
> Never extract archives from untrusted sources without prior inspection. It is possible that files are created outside of path, e.g. members that have absolute filenames starting with "/" or filenames with two dots "..". 

And I was convinced that we where using `extractmember` for this reason here: https://github.com/xenserver/planex/blob/depend-refactor/planex/tarball.py#L56

Looking back at the code... that was not the case. So I'll go on with your suggestion